datalad/interface/run_procedure.py

```diff
@@ -456,3 +464,47 @@ def __call__(
                 return_type='generator'
         ):
             yield r
+
+    @staticmethod
+    def custom_result_renderer(res, **kwargs):
+        from datalad.ui import ui
+        from datalad.interface.utils import default_result_renderer
+
+        if res['status'] != 'ok':
+            # logging complained about this already
+            return
+
+        if 'procedure' not in res.get('action', ''):
+            # it's not our business
+            default_result_renderer(res)
+            return
```

Comment:
This bit made me pause.  The previous 'status' guard ignores not "ok" results because logging has already complained about it.  But then here, we're deciding that a result doesn't concern us and using the default renderer.  That suggests that a non-ok result above could also not concern us, yet we're assuming it was logged.

Should this guard be move above the previous 'status' one?

Reply:
This might be true. I'm still a bit confused myself about the entire rendering/logging/yielding implications. 
But: As far as I'm aware, non-ok is always logged, so even if the results comes from an internal command call, it would be logged. That's not a matter of `run-procedure`'s implementation, except that it only applies if `run-procedure` actually yields those results from internal calls. So, throwing away non-ok first, seems right to me including results that are not 'procedure` results.

Reply:
As an example:

```
% datalad run-procedure --help-proc some
[WARNING] Cannot find procedure with name 'some' [run_procedure(some)] 
```

If you look into the code, you'll find, that `run_procedure` doesn't log that on its own. It's just yielding a result with that message.

Reply:
> But: As far as I'm aware, non-ok is always logged, so even if the results comes from an internal command call, it would be logged.

Indeed.  Sorry for the confusion.

>  So, throwing away non-ok first, seems right to me including results that are not 'procedure` results.

Agreed.